File: test/easyswift/NumberOfStepsToZeroTest.swift
import Testing

@testable import easy

@Test(arguments: [
  (2, 2),
  (0, 0),
  (1, 1),
  (8, 4),
  (14, 6),
  (123, 12),
  (1_000_000, 26),
  (-1, 0),
  (999_999, 31),
])
func numberOfStepsToZeroTest(num: Int, outcome: Int) {
  #expect(numberOfStepsToZero(num) == outcome)
}

File: test/easy/number_of_steps_to_zero_test.dart
import 'package:test/test.dart';
import 'package:dark_coding_challenges/easy/number_of_steps_to_zero.dart';

class NumberOfStepsToZeroTestCases {
  int n;
  int expected;
  NumberOfStepsToZeroTestCases(this.n, this.expected);
}

void main() {
  group("Feature:", () {
    final testCases = [
      NumberOfStepsToZeroTestCases(2, 2),
      NumberOfStepsToZeroTestCases(0, 0),
      NumberOfStepsToZeroTestCases(1, 1),
      NumberOfStepsToZeroTestCases(8, 4),
      NumberOfStepsToZeroTestCases(14, 6),
      NumberOfStepsToZeroTestCases(123, 12),
      NumberOfStepsToZeroTestCases(1_000_000, 26),
      NumberOfStepsToZeroTestCases(-1, 0),
      NumberOfStepsToZeroTestCases(999_999, 31),
    ];

    for (final step in testCases) {
      final n = step.n;
      final expected = step.expected;

      test("\n\tGiven: the integer $n.\n\tThen: it will take $expected steps to reduce it to zero.", () {
        int result = numberOfStepsToZero(n);
        expect(result, equals(expected));
      });
    }
  });
}
